### DISCUSSION TITLE
[Question] Dynamic resource reconfiguration during training

### DISCUSSION THREAD

**Discussion by fktmfpeps** [horovod/horovod#4045] [Q&A]
I found an interesting feature inside horovod which provides a way to dynamically scale up and down the number of workers during runtime, named as Elastic Horovod.
https://horovod.readthedocs.io/en/latest/elastic_include.html 

However, i also found that among the requirements, there also exist 'Gloo' installation requirements.
Does that mean i have to use 'Gloo' for communication even though i can use 'NCCL' as a communication primitive?
In other words, can i use 'NCCL' communication primitive while also using the elastic horovod feature?

[Edited in 0615]
I noticed that Gloo is a controller which should be compared with MPI, sincerely sorry for the misunderstood question above.
My question, again, then becomes that
Can we use Gloo with NCCL for dynamic resource reconfiguration during training?
